src/UI/Settings/SettingsLayout.js

```diff
@@ -248,5 +252,24 @@ module.exports = Marionette.Layout.extend({
         } else {
             $('body').removeClass('show-advanced-settings');
         }
+    },
+    
+    _setProxySettingsState : function() {
+        var value = Config.getValue(Config.Keys.ProxySettings);
+
+        if (value !== 'off') {
+            $('body').addClass('show-proxy-settings');
+        }
+    },
+    
+    _toggleProxySettings : function() {
+        console.log("Proxy Toggled")
+        var value = $('.x-proxy-settings').val();
+        Config.setValue(Config.Keys.ProxySettings, value);
+        if (value === 'off') {
+            $('body').removeClass('show-proxy-settings');
```

Comment:
Don't need to set a class at all, take a look at how we're doing it for auth:

https://github.com/Sonarr/Sonarr/blob/develop/src/UI/Settings/General/GeneralView.js#L60

Advanced settings applies to all settings pages (and edit modals) so we used a class.